src/wg/image/metadata.d

```diff
@@ -0,0 +1,63 @@
+module wg.image.metadata;
+
+import wg.image.imagebuffer;
+
+// metadata header struct
+
+struct MetaData
+{
+    char[4] magic;
+    uint bytes;
+    MetaData* next;
+
+    @property inout(void)[] data() inout return
+    {
+        // null? ie, bytes == 0? yes
+        return (cast(inout(void)*)&this)[typeof(this).sizeof .. typeof(this).sizeof + bytes];
+    }
+}
+
+inout(void)[] getMetadata(ref inout(ImageBuffer) image, const(char)[4] id)
+{
+    inout(MetaData)* md = image.metadata;
+    while (md)
+    {
+        if (md.magic == id)
+            return md.data;
+        md = md.next;
+    }
+    return null;
+}
+
+inout(MDType)* getMetadataAs(MDType)(ref inout(ImageBuffer) image)
+{
+    inout(void)[] md = image.getMetadata(MDType.ID);
+    if (!md)
+        return null;
+    assert(md.length >= MDType.sizeof); // shouldn't assert for this
+    return cast(inout(MDType)*)md.ptr;
+}
+
+
+// common metadata block for common but still somewhat niche data
+
+struct CommonMetadata
+{
+    enum char[4] ID = "META";
+
+    float pixelAspect = 1.0f; // defined as w/h; animorphic images > 1.0
+    float horizDpi = 0; // image has no associated physical units
+
+    float[2] getDPI() const { return [horizDpi, horizDpi / pixelAspect]; }
```

Comment:
How are we feeling about this?
I think it better expresses images that have no physical units associated; there's implicit knowledge available.

Reply:
The alternative way to express it with DPIx:DPIy, the DPI needs to have junk numbers to calculate the pixel aspect, even when no DPI exists.

Reply:
I really like your solution:
```
    float pixelAspect = 1.0f; // defined as w/h; animorphic images > 1.0
    float horizDpi = 0; // image has no associated physical units
```